### PR TITLE
chore: refactoring fetching result

### PR DESCRIPTION
**Pull request by kalombos**

### CODE REVIEW

**Comment by kalombos** [comment]
@Akerlay make review please. I will wait for your approve :)

File: peewee_async.py [hunk 1]
```diff
@@ -830,26 +735,42 @@ def execute_sql(self, *args, **kwargs):
                         (str(args), str(kwargs)))
         return super().execute_sql(*args, **kwargs)
 
-    async def aio_execute(self, query):
+    async def as_async_query_wrapper(self, cursor, query):
```
Comment (Akerlay) [inline peewee_async.py:738]: Maybe we shold make it classmethod of AsyncQueryWrapper? Something like await AsyncQueryWrapper.create(cursor, query)

File: peewee_async.py [hunk 2]
```diff
@@ -1359,21 +1247,66 @@ class AioQueryMixin:
     async def aio_execute(self, database):
         return await database.aio_execute(self)
 
+    async def as_async_query_wrapper(self, cursor):
```
Comment (Akerlay) [inline peewee_async.py:1250]: Same as above, it may be classmethod of AsyncQueryWrapper

File: peewee_async.py [hunk 3]
```diff
@@ -1359,21 +1247,66 @@ class AioQueryMixin:
...
+        return result
 
 
 class AioModelSelect(peewee.ModelSelect, AioQueryMixin):
 
+    async def fetch_results(self, cursor):
+        return await self.as_async_query_wrapper(cursor)
+
+    @peewee.database_required
+    async def aio_scalar(self, database, as_tuple=False, as_dict=False):
```
Comment (Akerlay) [inline peewee_async.py:1295]: as_dict not used?

File: peewee_async.py [hunk 4]
```diff
@@ -1359,21 +1247,66 @@ class AioQueryMixin:
...
+        return cursor.rowcount
 
 
 class AioModelInsert(peewee.ModelInsert, AioQueryMixin):
-    pass
+    async def fetch_results(self, cursor):
+        if self._returning is not None and len(self._returning) > 1:
+            return await self.as_async_query_wrapper(cursor=cursor)
+
+        if self._returning:
```
Comment (Akerlay) [inline peewee_async.py:1276]:
```
if self._returning:
    row = await cursor.fetchone()
    return  row and row[0]   # may be a bit complicated we can also keep original, but with returns, not result = ...
else:
    return await self._database.last_insert_id_async(cursor)
```

more

File: peewee_async.py (hunk 3 above)
Comment (kalombos) [reply peewee_async.py:1295]: I thought the deprecated scalar had had it either. Actually i copy-pasted it from sync peewee and i think we will add this param lately. But i will remove it for the current state. Thank you

**Pull request merged**